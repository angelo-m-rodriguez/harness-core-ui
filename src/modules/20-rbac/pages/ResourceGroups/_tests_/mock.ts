/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ResponsePageResourceGroupV2Response } from 'services/resourcegroups'

export const resourceGroupListResponse: ResponsePageResourceGroupV2Response = {
  status: 'SUCCESS',
  data: {
    totalPages: 2,
    totalItems: 20,
    pageItemCount: 10,
    pageSize: 10,
    content: [
      {
        resourceGroup: {
          accountIdentifier: 'kmpySmUISimoRrJL6NL73w',
          identifier: 'ss33',
          name: 'ss33',
          color: '#0063f7',
          includedScopes: [
            {
              filter: 'EXCLUDING_CHILD_SCOPES',
              accountIdentifier: 'kmpySmUISimoRrJL6NL73w'
            }
          ],
          resourceFilter: {
            includeAllResources: false,
            resources: [{ resourceType: 'ORGANIZATION' }, { resourceType: 'SECRET_MANAGER' }]
          },
          tags: {},
          description: ''
        },
        createdAt: 1614237155621,
        harnessManaged: false,
        lastModifiedAt: 1614237168823
      },
      {
        resourceGroup: {
          accountIdentifier: 'kmpySmUISimoRrJL6NL73w',
          identifier: 'starte',
          name: 'starte',
          color: '#0063f7',
          includedScopes: [
            {
              filter: 'EXCLUDING_CHILD_SCOPES',
              accountIdentifier: 'kmpySmUISimoRrJL6NL73w'
            }
          ],
          resourceFilter: {
            includeAllResources: false,
            resources: [
              { resourceType: 'ORGANIZATION' },
              { resourceType: 'SECRET_MANAGER' },
              { resourceType: 'PROJECT' }
            ]
          },
          tags: {},
          description: ''
        },
        createdAt: 1614235877442,
        harnessManaged: false,
        lastModifiedAt: 1614237139987
      },
      {
        resourceGroup: {
          accountIdentifier: 'kmpySmUISimoRrJL6NL73w',

          identifier: 'swarfdfds34343',
          name: 'swarfdfds34343',
          color: '#0063f7',
          includedScopes: [
            {
              filter: 'EXCLUDING_CHILD_SCOPES',
              accountIdentifier: 'kmpySmUISimoRrJL6NL73w'
            }
          ],
          resourceFilter: {
            includeAllResources: false,
            resources: [{ resourceType: 'ORGANIZATION' }, { resourceType: 'PROJECT' }]
          },

          tags: {},
          description: ''
        },
        createdAt: 1614235749097,
        harnessManaged: false,
        lastModifiedAt: 1614235755919
      }
    ],
    pageIndex: 0,
    empty: false
  },
  correlationId: '74bd725c-de9e-47eb-b588-c9aca8b1433c'
}
