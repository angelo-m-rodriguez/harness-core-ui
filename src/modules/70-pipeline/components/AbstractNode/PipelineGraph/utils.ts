import type { RefObject } from 'react'
import type { StageElementWrapperConfig } from 'services/cd-ng'
const INITIAL_ZOOM_LEVEL = 1
const ZOOM_INC_DEC_LEVEL = 0.1
const getFinalSVGArrowPath = (parentId: string, id1 = '', id2 = '', isParallelNode = false): string => {
  if (!parentId) {
    return ''
  }
  const node1 = getComputedPosition(parentId, id1)
  const node2 = getComputedPosition(parentId, id2)
  if (!node1 || !node2) {
    return ''
  }

  const node1VerticalMid = node1.top + node1.height / 2
  const node2VerticalMid = node2.top + node2.height / 2

  const startPoint = `${node1.right},${node1VerticalMid}`
  const horizontalMid = Math.abs((node1.right + node2.left) / 2)
  const endPoint = `${node2.left},${node2VerticalMid}`

  if (node2.y < node1.y) {
    //  child node is at top
    const curveLeftToTop = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid - 20}`
    const curveBottomToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToTop} 
    L${horizontalMid},${node2VerticalMid + 20} ${curveBottomToRight} L${endPoint}`
  } else if (node1.y === node2.y) {
    // both nodes are at same level vertically
    return `M${startPoint}  L${endPoint} `
  } else {
    //  child node is at bottom
    const curveLeftToBottom = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid + 20}`

    const curveTopToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`

    if (isParallelNode) {
      const updatedStart = node1.left - 45 // new start point
      const parallelLinkStart = `${
        updatedStart // half of link length
      },${node1VerticalMid}`

      const curveLBparallel = `Q${updatedStart + 20},${node1VerticalMid} ${updatedStart + 20},${node1VerticalMid + 20} `
      const curveTRparallel = `Q${updatedStart + 20},${node2VerticalMid} ${updatedStart + 40},${node2VerticalMid}`

      return `M${parallelLinkStart} 
      ${curveLBparallel} 
      L${updatedStart + 20},${node2VerticalMid - 20} 
      ${curveTRparallel} 
      L${node2.left},${node2VerticalMid} 
      M${node2.right},${node2VerticalMid} 
      L${node2.right + 10},${node2VerticalMid}
      Q${node2.right + 25},${node2VerticalMid} ${node2.right + 25},${node2VerticalMid - 20}      
      L${node2.right + 25},${node1VerticalMid + 20}
      Q${node2.right + 25},${node1VerticalMid} ${node2.right + 40},${node1VerticalMid}
      `
    }
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToBottom} 
    L${horizontalMid},${node2VerticalMid - 20} ${curveTopToRight} L${endPoint}`
  }
}

const getComputedPosition = (parentId: string, childId: string): DOMRect => {
  const parentPos = document.getElementById(parentId)?.getBoundingClientRect() as DOMRect
  const childPos = document.getElementById(childId)?.getBoundingClientRect() as DOMRect
  const updatedTop = childPos.top - parentPos.top
  const updatedLeft = childPos.left - parentPos.left
  const updatedRight = updatedLeft + childPos.width
  const updatedBottom = updatedTop + childPos.height
  const updatedPos: DOMRect = {
    ...childPos,
    left: updatedLeft,
    top: updatedTop,
    right: updatedRight,
    bottom: updatedBottom,
    width: childPos.width,
    height: childPos.height,
    x: childPos.x,
    y: childPos.y
  }
  return updatedPos
}

const setupDragEventListeners = (canvasRef: RefObject<HTMLDivElement>): (() => void) => {
  let offset = [0, 0]
  const divOverlay = (canvasRef.current as HTMLDivElement).parentElement as HTMLDivElement
  const canvas = canvasRef.current as HTMLDivElement
  let isDown = false

  const onMouseMove = (e: any): void => {
    e.preventDefault()
    if (isDown) {
      canvas.style.left = e.clientX + offset[0] + 'px'
      canvas.style.top = e.clientY + offset[1] + 'px'
    }
  }

  const onMouseDown = (e: any): void => {
    isDown = true
    offset = [canvas.offsetLeft - e.clientX, canvas.offsetTop - e.clientY]
  }
  const onMouseUp = (): void => {
    isDown = false
  }

  divOverlay.addEventListener('mousedown', onMouseDown, true)
  document.addEventListener('mouseup', onMouseUp, true)
  document.addEventListener('mousemove', onMouseMove, true)
  return () => {
    divOverlay.removeEventListener('mousedown', onMouseDown)
    document.removeEventListener('mouseup', onMouseUp)
    document.removeEventListener('mousemove', onMouseMove)
  }
}

const getSVGLinksFromPipeline = (stages?: StageElementWrapperConfig[], resultArr: string[] = []): string[] => {
  let prevElement: StageElementWrapperConfig
  stages?.forEach(stage => {
    if (!prevElement) {
      prevElement = stage
      return
    }
    if (stage.parallel) {
      getParallelNodeLinks(stage.parallel, prevElement, resultArr)
    } else {
      resultArr.push(
        getFinalSVGArrowPath('tree-container', prevElement.stage?.identifier as string, stage.stage?.identifier, false)
      )
    }
    prevElement = stage
  })
  return resultArr
}
const getParallelNodeLinks = (
  stages: StageElementWrapperConfig[],
  prevStage: StageElementWrapperConfig,
  resultArr: string[] = []
): void => {
  const [firstStage, ...restStages] = stages
  resultArr.push(
    getFinalSVGArrowPath('tree-container', prevStage.stage?.identifier as string, firstStage.stage?.identifier, false)
  )
  restStages?.forEach(stage => {
    resultArr.push(
      getFinalSVGArrowPath('tree-container', firstStage.stage?.identifier as string, stage.stage?.identifier, true)
    )
  })
}

const getScaleToFitValue = (elm: HTMLElement, paddingFromBottom = 20): number => {
  return (
    1 /
    Math.max(
      elm.clientWidth / window.innerWidth,
      elm.clientHeight / (window.innerHeight - elm.offsetTop - paddingFromBottom)
    )
  )
}
export {
  getFinalSVGArrowPath,
  setupDragEventListeners,
  getSVGLinksFromPipeline,
  ZOOM_INC_DEC_LEVEL,
  INITIAL_ZOOM_LEVEL,
  getScaleToFitValue
}
