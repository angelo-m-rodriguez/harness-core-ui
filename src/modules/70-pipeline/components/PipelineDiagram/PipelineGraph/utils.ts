import { defaultTo } from 'lodash-es'
import type { PipelineGraphState } from '../types'

const shouldAttachRef = (
  intersectingIndex: number,
  isCurrentChildLast: boolean,
  indexRelativeToParent: number
): boolean => {
  return intersectingIndex === -1 // intersecting index is not set
    ? isCurrentChildLast // current child is not the last child in parallel
    : intersectingIndex === indexRelativeToParent // intersecting index is not equal to indexRelativeToParent (i.e index+1 as parent is 0)
}

const shouldRenderGroupNode = (attachRef: boolean, isCurrentChildLast: boolean): boolean => {
  return attachRef && !isCurrentChildLast // render group node is ref is attached and current node is not the last child
}

const isFirstNodeAGroupNode = (intersectingIndex: number, collapseOnIntersect = false, childLength = 0): boolean =>
  intersectingIndex === 0 && collapseOnIntersect && childLength > 0 // if intersecting index is 0 and collapseOnIntersect is enabled

const showChildNode = (indexRelativeToParent: number, intersectingIndex: number): boolean => {
  return !(indexRelativeToParent > intersectingIndex && intersectingIndex !== -1) // child index is greater than intersectingIndex and intersectingIndex is set to some value
}

const isNodeParallel = (node: PipelineGraphState): boolean => defaultTo(node?.children?.length, 0) > 0

export { shouldAttachRef, shouldRenderGroupNode, isFirstNodeAGroupNode, showChildNode, isNodeParallel }
