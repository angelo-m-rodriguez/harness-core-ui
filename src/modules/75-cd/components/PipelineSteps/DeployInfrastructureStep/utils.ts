/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType, SelectOption } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface PipelineInfrastructureV2 extends PipelineInfrastructure {
  environmentOrEnvGroupRef?: SelectOption
  environmentGroup?: any
  environmentRef2?: any
  infrastructureRef?: any
}

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructureV2
  onUpdate?: (data: PipelineInfrastructureV2) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructureV2
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructureV2>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructureV2): boolean {
  if (
    getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.environmentOrEnvGroupRef)
  ) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
