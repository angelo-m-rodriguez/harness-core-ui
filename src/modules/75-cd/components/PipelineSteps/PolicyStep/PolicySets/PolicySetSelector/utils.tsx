/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { FontVariation, Layout, Text } from '@harness/uicore'
import { useStrings } from 'framework/strings'

export const PolicySetSelectorDialogTitle = (): JSX.Element => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal
      spacing="xsmall"
      padding={{ top: 'xxlarge', left: 'medium', right: 'large', bottom: 'medium' }}
      flex={{ justifyContent: 'space-between' }}
    >
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H3 }}>{getString('common.policiesSets.selectPolicySet')}</Text>
      </Layout.Vertical>
    </Layout.Horizontal>
  )
}
