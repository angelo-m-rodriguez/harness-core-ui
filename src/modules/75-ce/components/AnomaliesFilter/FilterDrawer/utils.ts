/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@harness/uicore'
import { omit } from 'lodash-es'
import type { AnomalyFilterProperties } from 'services/ce'
import type { AnomaliesFilterFormType } from './FilterDrawer'

const getMultiSelectOptions = (values: string[]): MultiSelectOption[] =>
  values.map(item => ({ label: item, value: item }))

const getValueFromOption = (values: MultiSelectOption[]): string[] => values.map(item => item.value.toString())

export const getAnomalyFormValuesFromFilterProperties = (
  filterProperties: AnomalyFilterProperties
): AnomaliesFilterFormType => {
  const formValues: AnomaliesFilterFormType = {}

  Object.entries(omit(filterProperties, 'filterType')).forEach(([key, value]) => {
    if (value) {
      if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
        formValues[key] = +value
      } else {
        ;(formValues as any)[key] = getMultiSelectOptions(value as string[])
      }
    }
  })

  return formValues
}

export const getAnomalyFilterPropertiesFromForm = (formData: AnomaliesFilterFormType): AnomalyFilterProperties => {
  const filterProperties: AnomalyFilterProperties = { filterType: 'Anomaly' }

  Object.entries(formData).forEach(([key, value]) => {
    if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
      filterProperties[key] = +value
    } else {
      ;(filterProperties as any)[key] = getValueFromOption(value)
    }
  })

  return filterProperties
}
