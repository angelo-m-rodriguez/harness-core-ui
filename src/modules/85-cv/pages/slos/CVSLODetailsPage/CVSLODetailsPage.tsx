/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { useParams, useHistory } from 'react-router-dom'
import { Container, Page, Tabs } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import { useGetSLODetails } from 'services/cv'
import routes from '@common/RouteDefinitions'
import { useQueryParams } from '@common/hooks'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import { getSearchString } from '@cv/utils/CommonUtils'
import CVCreateSLO from '@cv/pages/slos/components/CVCreateSLO/CVCreateSLO'
import HeaderTitle from './views/HeaderTitle'
import { SLODetailsPageTabIds } from './CVSLODetailsPage.types'
import css from './CVSLODetailsPage.module.scss'

const CVSLODetailsPage: React.FC = () => {
  const history = useHistory()
  const { getString } = useStrings()
  const { accountId, orgIdentifier, projectIdentifier, identifier } = useParams<
    ProjectPathProps & { identifier: string }
  >()
  const { tab = SLODetailsPageTabIds.Configurations, monitoredServiceIdentifier } =
    useQueryParams<{ tab?: SLODetailsPageTabIds; monitoredServiceIdentifier?: string }>()

  const { data, loading } = useGetSLODetails({
    identifier,
    queryParams: {
      accountId,
      orgIdentifier,
      projectIdentifier
    }
  })

  /* istanbul ignore next */
  const onTabChange = (nextTab: SLODetailsPageTabIds): void => {
    if (nextTab !== tab) {
      history.push({
        pathname: routes.toCVSLODetailsPage({
          identifier,
          accountId,
          orgIdentifier,
          projectIdentifier
        }),
        search: getSearchString({ tab: nextTab, monitoredServiceIdentifier })
      })
    }
  }

  const { description, sloDashboardWidget } = data?.data ?? {}
  const breadcrumbLinks = [
    {
      url: routes.toCVSLOs({ accountId, orgIdentifier, projectIdentifier }),
      label: getString('cv.SLO')
    }
  ]

  return (
    <div>
      <Page.Header
        size="large"
        title={<HeaderTitle loading={loading} title={sloDashboardWidget?.title} description={description} />}
        breadcrumbs={<NGBreadcrumbs links={breadcrumbLinks} />}
      />
      <Container className={css.tabContainer}>
        <Tabs
          id="slo-details-page-tabs"
          selectedTabId={tab}
          onChange={onTabChange}
          tabList={[
            {
              id: SLODetailsPageTabIds.Details,
              title: 'Details',
              disabled: true
            },
            {
              id: SLODetailsPageTabIds.Configurations,
              title: getString('cv.monitoredServices.monitoredServiceTabs.configurations'),
              panel: <CVCreateSLO />
            }
          ]}
        ></Tabs>
      </Container>
    </div>
  )
}

export default CVSLODetailsPage
