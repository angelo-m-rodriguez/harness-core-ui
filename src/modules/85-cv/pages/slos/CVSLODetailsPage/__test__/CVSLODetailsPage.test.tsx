/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, RenderResult } from '@testing-library/react'
import * as cvServices from 'services/cv'
import { TestWrapper } from '@common/utils/testUtils'
import CVSLODetailsPage from '../CVSLODetailsPage'
import { responseSLODashboardDetail, testWrapperProps } from './CVSLODetailsPage.mock'

jest.mock('@cv/pages/slos/components/CVCreateSLO/CVCreateSLO', () => ({
  __esModule: true,
  default: function CVCreateSLO() {
    return <div>MOCKED - CVCreateSLO</div>
  }
}))

const renderComponent = (): RenderResult => {
  return render(
    <TestWrapper {...testWrapperProps}>
      <CVSLODetailsPage />
    </TestWrapper>
  )
}

describe('Test cases for CVSLODetailsPage', () => {
  test('it should render the component and take a snapshot', () => {
    jest
      .spyOn(cvServices, 'useGetSLODetails')
      .mockReturnValue({ data: responseSLODashboardDetail, loading: false } as any)

    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  test('it should handle the loading state', () => {
    jest.spyOn(cvServices, 'useGetSLODetails').mockReturnValue({ data: null, loading: true } as any)

    const { container } = renderComponent()

    expect(container.getElementsByClassName('bp3-skeleton')).toHaveLength(2)
  })
})
