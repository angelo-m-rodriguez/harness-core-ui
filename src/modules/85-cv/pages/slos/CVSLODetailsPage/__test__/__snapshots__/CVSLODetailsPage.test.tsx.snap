// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for CVSLODetailsPage it should render the component and take a snapshot 1`] = `
<div>
  <div>
    <div
      class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--large StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="PageHeader--breadcrumbsDiv"
        >
          <div
            class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/ACCOUNT_ID/cv/orgs/ORG_IDENTIFIER/projects/PROJECT_IDENTIFIER"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                  data-icon="cv-main"
                >
                  <svg
                    height="16"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M25.933 24.357A12.972 12.972 0 0116 28.969a12.972 12.972 0 01-9.933-4.612l6.025-3.834A5.976 5.976 0 0016 21.969a5.977 5.977 0 003.907-1.446l6.026 3.834zm1.066-1.455l-5.906-3.759a6.001 6.001 0 00-4.193-9.107V3C23.66 3.462 29 9.092 29 15.97a12.94 12.94 0 01-2 6.932zM15.1 3C8.34 3.462 3 9.092 3 15.97c0 2.548.733 4.925 2 6.932l5.907-3.759a6.001 6.001 0 014.192-9.107V3z"
                      fill="url(#cv-main_svg__paint0_linear_20630_159361)"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M19.558 13.286a.9.9 0 01-.044 1.272l-4.286 4a.9.9 0 01-1.228 0l-1.714-1.6a.9.9 0 011.228-1.316l1.1 1.027 3.672-3.427a.9.9 0 011.272.044z"
                      fill="#AE82FC"
                      fill-rule="evenodd"
                    />
                    <defs>
                      <lineargradient
                        gradientUnits="userSpaceOnUse"
                        id="cv-main_svg__paint0_linear_20630_159361"
                        x1="7.5"
                        x2="20.926"
                        y1="3"
                        y2="28.987"
                      >
                        <stop
                          stop-color="#BEA1FF"
                        />
                        <stop
                          offset="1"
                          stop-color="#8A36FF"
                        />
                      </lineargradient>
                    </defs>
                  </svg>
                </span>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  PROJECT_IDENTIFIER
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/ACCOUNT_ID/cv/orgs/ORG_IDENTIFIER/projects/PROJECT_IDENTIFIER/slos"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  cv.SLO
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-evenly StyledProps--flex-alignItems-flex-start"
          style="height: 45px;"
        >
          <h3
            class="StyledProps--font StyledProps--font-h3 StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-h5"
            level="3"
          >
            cv.SLOWithName
          </h3>
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey600 StyledProps--font-variation-body2"
          >
            99.9 percent of logins should have &lt;= 50ms latency
          </p>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main tabContainer"
    >
      <div
        class="bp3-tabs Tabs--main Tabs--horizontal"
      >
        <div
          class="bp3-tab-list"
          role="tablist"
        >
          <div
            class="bp3-tab-indicator-wrapper"
            style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
          >
            <div
              class="bp3-tab-indicator"
            />
          </div>
          <div
            aria-controls="bp3-tab-panel_slo-details-page-tabs_Details"
            aria-disabled="true"
            aria-expanded="false"
            aria-selected="false"
            class="bp3-tab"
            data-tab-id="Details"
            id="bp3-tab-title_slo-details-page-tabs_Details"
            role="tab"
          >
            Details
          </div>
          <div
            aria-controls="bp3-tab-panel_slo-details-page-tabs_Configurations"
            aria-disabled="false"
            aria-expanded="true"
            aria-selected="true"
            class="bp3-tab"
            data-tab-id="Configurations"
            id="bp3-tab-title_slo-details-page-tabs_Configurations"
            role="tab"
            tabindex="0"
          >
            cv.monitoredServices.monitoredServiceTabs.configurations
          </div>
        </div>
        <div
          aria-hidden="false"
          aria-labelledby="bp3-tab-title_slo-details-page-tabs_Configurations"
          class="bp3-tab-panel"
          id="bp3-tab-panel_slo-details-page-tabs_Configurations"
          role="tabpanel"
        >
          <div>
            MOCKED - CVCreateSLO
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
